xlive/Blam/Engine/render/render.cpp

```diff
@@ -304,6 +348,428 @@ void __cdecl render_window(window_bound* window, bool is_texture_camera)
     return;
 }
 
+void __cdecl render_scene(
+    int32 render_layer_debug_view,
+    bool render_transparent_geo,
+    bool lens_flare_occlusion_test,
+    bool render_layer_selfibloomination,
+    int32 hologram_flag,
+    int32 effect_flag,
+    real32 depth_range)
+{
+    s_render_scene_parameters parameters;
+    uint32* g_scene_rendered_count = global_scene_rendered_count_get();
+    parameters.scene_rendered_count = ++*g_scene_rendered_count;
+    parameters.effect_flags = effect_flag;
+    parameters.depth_range = depth_range;
+
+    
+    rasterizer_dx9_render_scene_start(&parameters);
+
+    // Sky rendering
+    const s_frame* global_window_parameters = global_window_parameters_get();
+
+    create_visible_render_primitives(hologram_flag);
+
+    const real32 dr = global_window_parameters->camera.z_far - global_window_parameters->camera.z_near;
+    const real_vector4d constants { 0.f, 1.f, dr, 1.f };
+    IDirect3DDevice9Ex* global_d3d_device = rasterizer_dx9_device_get_interface();
+    if (rasterizer_get_main_pixel_shader_cache()->test_cache(16, &constants, 1))
+    {
+        global_d3d_device->SetPixelShaderConstantF(16, (real32*)&constants, 1);
+    }
+
+    if (render_layer_debug_view != 5)
+    {
+        if (render_layer_debug_view == 1 || render_layer_debug_view == 2)
+        {
+            s_rasterizer_globals* rasterizer_globals = rasterizer_globals_get();
+            if (rasterizer_globals->d3d9_sm3_supported)
+            {
+                *global_rasterizer_pixel_shader_index_get() = 2;
+                g_dx9_dont_draw_to_depth_target_if_mrt_is_used = false;
+            }
+
+            rasterizer_dx9_perf_event_begin("texaccum", NULL);
+            if (prepare_render_layer(_render_layer_texture_accumulate))
+            {
+                draw_render_layer();
+                reset_after_render_layer_draw();
+            }
```

Comment:
function instead? this sequence is quite repetitive

Reply:
added the DRAW_RENDER_LAYER macro